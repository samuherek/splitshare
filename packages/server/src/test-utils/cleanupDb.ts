import { getConnection } from 'typeorm';

export default async function cleanupDb() {
  await getConnection().query('DELETE FROM bill_user');
  await getConnection().query('DELETE FROM receipt');
  await getConnection().query('DELETE FROM bill_invite');
  await getConnection().query('DELETE FROM bill');
  await getConnection().query('DELETE FROM "user"');
}
