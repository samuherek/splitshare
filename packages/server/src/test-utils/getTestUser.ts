import { Connection } from 'typeorm';
import { User } from '../features/user/entity';

export const testUser = {
  email: 'user@email.com',
};

export async function getTestUser(conn: Connection) {
  const user = await conn.getRepository(User).findOne(testUser);

  if (!user) {
    throw new Error('No Test user found');
  }

  return user;
}
