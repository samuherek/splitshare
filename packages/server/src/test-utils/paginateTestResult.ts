import { PaginationConnection } from '../types';

export default function paginateTestResult(
  resArray: any[]
): PaginationConnection<any> {
  return {
    edges: resArray.map((r) => ({ node: r, cursor: 'nodeCursor' })),
    pageInfo: {
      hasNextPage: false,
      endCursor: 'endCursor',
      itemsCount: 0,
    },
  };
}
