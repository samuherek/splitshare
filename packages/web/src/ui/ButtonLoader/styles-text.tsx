import styled from 'styled-components';

export const Base = styled.span``;
