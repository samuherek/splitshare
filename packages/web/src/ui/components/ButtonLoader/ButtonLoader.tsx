import capitalize from 'capitalize';
import clsx from 'clsx';
import React from 'react';
import styled, { keyframes } from 'styled-components';
import {
  DarkContained,
  PrimaryContained,
  SecondaryContained,
  TertiaryContained,
} from './styles-contained';
import {
  DarkText,
  PrimaryText,
  SecondaryText,
  TertiaryText,
} from './styles-text';

interface Props {
  color?: 'primary' | 'secondary' | 'tertiary' | 'dark';
  variant?: 'text' | 'contained';
}

export const classes = {
  root: 'ButtonLoader',
  primary: 'primary',
};

const stretch = keyframes`
  0% {
    transform: scale(0.5);
    opacity: 0.3;
  }
  50% {
    opacity: 0.6;
  }
  100% {
    transform: scale(1);
    opacity: 1;
  }
`;

const ButtonLoaderStyled = styled.span`
  display: flex;
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);

  i {
    animation-direction: alternate;
    animation-duration: 0.5s;
    animation-fill-mode: none;
    animation-iteration-count: infinite;
    animation-name: ${stretch};
    animation-play-state: running;
    animation-timing-function: ease-out;
    border-radius: 100%;
    display: block;
    height: 6px;
    width: 6px;
    margin: 0 3px;

    &:nth-child(1) {
      animation-delay: 0.1s;
    }

    &:nth-child(2) {
      animation-delay: 0.3s;
    }

    &:nth-child(3) {
      animation-delay: 0.5s;
    }
  }
`;

const themedComponent = {
  primaryText: PrimaryText,
  primaryContained: PrimaryContained,
  secondaryText: SecondaryText,
  secondaryContained: SecondaryContained,
  tertiaryText: TertiaryText,
  tertiaryContained: TertiaryContained,
  darkText: DarkText,
  darkContained: DarkContained,
};

function ButtonLoader({ color = 'primary', variant = 'text' }: Props = {}) {
  const key = `${color}${capitalize(variant)}`;
  // @ts-ignore
  const ThemedComponent = themedComponent[key] || ButtonLoaderStyled;

  return (
    <ButtonLoaderStyled
      as={ThemedComponent}
      className={clsx(classes.root, {
        [classes.primary]: color === 'primary',
      })}
    >
      <i />
      <i />
      <i />
    </ButtonLoaderStyled>
  );
}

export default ButtonLoader;
