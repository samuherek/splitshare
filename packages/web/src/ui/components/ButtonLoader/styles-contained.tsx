import styled from 'styled-components';

// #0ec76a
export const PrimaryContained = styled.span`
  i {
  }
`;

// #ffffff
export const SecondaryContained = styled.span`
  i {
  }
`;

export const TertiaryContained = styled.span`
  i {
  }
`;

export const DarkContained = styled.span`
  i {
  }
`;
