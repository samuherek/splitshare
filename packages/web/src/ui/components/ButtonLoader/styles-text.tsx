import styled from 'styled-components';

// #0ec76a
export const PrimaryText = styled.span`
  i {
  }
`;

// #ffffff
export const SecondaryText = styled.span`
  i {
  }
`;

export const TertiaryText = styled.span`
  i {
  }
`;

export const DarkText = styled.span`
  i {
  }
`;
