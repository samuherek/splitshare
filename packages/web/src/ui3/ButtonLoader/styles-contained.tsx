import styled from 'styled-components';

export const Base = styled.span``;
// // #0ec76a
// export const PrimaryContained = styled.span`
//   i {
//     background: ${colorByKey("white")};
//   }
// `;

// // #ffffff
// export const SecondaryContained = styled.span`
//   i {
//     background: ${colorByKey("blueyGrey")};
//   }
// `;

// export const TertiaryContained = styled.span`
//   i {
//     background: ${colorByKey("cloudyBlue")};
//   }
// `;

// export const DarkContained = styled.span`
//   i {
//     background: ${colorByKey("cloudyBlue")};
//   }
// `;
