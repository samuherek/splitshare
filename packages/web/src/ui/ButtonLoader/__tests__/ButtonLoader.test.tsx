import React from "react";
import { render } from "../../../test-utils/test-utils";
import ButtonLoader, { classes } from "../ButtonLoader";

test("should match snapshots", () => {
  const defaultLoader = render(<ButtonLoader />);
  expect(defaultLoader.container.firstChild).toMatchSnapshot();

  const primaryTextButton = render(<ButtonLoader color="primary" />);
  expect(primaryTextButton.container.firstChild).toMatchSnapshot();

  const secondaryTextButton = render(<ButtonLoader color="secondary" />);
  expect(secondaryTextButton.container.firstChild).toMatchSnapshot();

  const tertiaryTextButton = render(<ButtonLoader color="tertiary" />);
  expect(tertiaryTextButton.container.firstChild).toMatchSnapshot();

  const primaryContainedButton = render(
    <ButtonLoader color="primary" variant="contained" />
  );
  expect(primaryContainedButton.container.firstChild).toMatchSnapshot();

  const secondaryContainedButton = render(
    <ButtonLoader color="secondary" variant="contained" />
  );
  expect(secondaryContainedButton.container.firstChild).toMatchSnapshot();

  const tertiaryContainedButton = render(
    <ButtonLoader color="tertiary" variant="contained" />
  );
  expect(tertiaryContainedButton.container.firstChild).toMatchSnapshot();
});

test("should render with the default styles", () => {
  const { container } = render(<ButtonLoader />);
  const loader = container.firstChild;

  expect(loader).toHaveClass(classes.root);
  expect(loader).toHaveProperty("nodeName", "SPAN");
  // @ts-ignore
  expect(loader).toHaveStyleRule("display", "flex");
  // @ts-ignore
  expect(loader).toHaveStyleRule("position", "absolute");
  // @ts-ignore
  expect(loader).toHaveStyleRule("top", "50%");
  // @ts-ignore
  expect(loader).toHaveStyleRule("left", "50%");
  // @ts-ignore
  expect(loader).toHaveStyleRule("transform", "translate(-50%,-50%)");
});

test("should render with the loader icon", () => {
  const { container } = render(<ButtonLoader />);
  const icons = container.querySelectorAll("i");

  expect(icons.length).toBe(3);
});
