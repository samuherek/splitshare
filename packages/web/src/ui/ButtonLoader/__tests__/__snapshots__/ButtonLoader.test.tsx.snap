// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshots 1`] = `
.c0 i {
  background: #0ec76a;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
}

.c1 i {
  -webkit-animation-direction: alternate;
  animation-direction: alternate;
  -webkit-animation-duration: 0.5s;
  animation-duration: 0.5s;
  -webkit-animation-fill-mode: none;
  animation-fill-mode: none;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-name: lfUrxy;
  animation-name: lfUrxy;
  -webkit-animation-play-state: running;
  animation-play-state: running;
  -webkit-animation-timing-function: ease-out;
  animation-timing-function: ease-out;
  border-radius: 100%;
  display: block;
  height: 6px;
  width: 6px;
  margin: 0 3px;
}

.c1 i:nth-child(1) {
  -webkit-animation-delay: 0.1s;
  animation-delay: 0.1s;
}

.c1 i:nth-child(2) {
  -webkit-animation-delay: 0.3s;
  animation-delay: 0.3s;
}

.c1 i:nth-child(3) {
  -webkit-animation-delay: 0.5s;
  animation-delay: 0.5s;
}

<span
  class="c0 c1 ButtonLoader primary"
>
  <i />
  <i />
  <i />
</span>
`;

exports[`should match snapshots 2`] = `
.c0 i {
  background: #0ec76a;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
}

.c1 i {
  -webkit-animation-direction: alternate;
  animation-direction: alternate;
  -webkit-animation-duration: 0.5s;
  animation-duration: 0.5s;
  -webkit-animation-fill-mode: none;
  animation-fill-mode: none;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-name: lfUrxy;
  animation-name: lfUrxy;
  -webkit-animation-play-state: running;
  animation-play-state: running;
  -webkit-animation-timing-function: ease-out;
  animation-timing-function: ease-out;
  border-radius: 100%;
  display: block;
  height: 6px;
  width: 6px;
  margin: 0 3px;
}

.c1 i:nth-child(1) {
  -webkit-animation-delay: 0.1s;
  animation-delay: 0.1s;
}

.c1 i:nth-child(2) {
  -webkit-animation-delay: 0.3s;
  animation-delay: 0.3s;
}

.c1 i:nth-child(3) {
  -webkit-animation-delay: 0.5s;
  animation-delay: 0.5s;
}

<span
  class="c0 c1 ButtonLoader primary"
>
  <i />
  <i />
  <i />
</span>
`;

exports[`should match snapshots 3`] = `
.c0 i {
  background: #a0a2ad;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
}

.c1 i {
  -webkit-animation-direction: alternate;
  animation-direction: alternate;
  -webkit-animation-duration: 0.5s;
  animation-duration: 0.5s;
  -webkit-animation-fill-mode: none;
  animation-fill-mode: none;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-name: lfUrxy;
  animation-name: lfUrxy;
  -webkit-animation-play-state: running;
  animation-play-state: running;
  -webkit-animation-timing-function: ease-out;
  animation-timing-function: ease-out;
  border-radius: 100%;
  display: block;
  height: 6px;
  width: 6px;
  margin: 0 3px;
}

.c1 i:nth-child(1) {
  -webkit-animation-delay: 0.1s;
  animation-delay: 0.1s;
}

.c1 i:nth-child(2) {
  -webkit-animation-delay: 0.3s;
  animation-delay: 0.3s;
}

.c1 i:nth-child(3) {
  -webkit-animation-delay: 0.5s;
  animation-delay: 0.5s;
}

<span
  class="c0 c1 ButtonLoader"
>
  <i />
  <i />
  <i />
</span>
`;

exports[`should match snapshots 4`] = `
.c0 i {
  background: #b7bac7;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
}

.c1 i {
  -webkit-animation-direction: alternate;
  animation-direction: alternate;
  -webkit-animation-duration: 0.5s;
  animation-duration: 0.5s;
  -webkit-animation-fill-mode: none;
  animation-fill-mode: none;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-name: lfUrxy;
  animation-name: lfUrxy;
  -webkit-animation-play-state: running;
  animation-play-state: running;
  -webkit-animation-timing-function: ease-out;
  animation-timing-function: ease-out;
  border-radius: 100%;
  display: block;
  height: 6px;
  width: 6px;
  margin: 0 3px;
}

.c1 i:nth-child(1) {
  -webkit-animation-delay: 0.1s;
  animation-delay: 0.1s;
}

.c1 i:nth-child(2) {
  -webkit-animation-delay: 0.3s;
  animation-delay: 0.3s;
}

.c1 i:nth-child(3) {
  -webkit-animation-delay: 0.5s;
  animation-delay: 0.5s;
}

<span
  class="c0 c1 ButtonLoader"
>
  <i />
  <i />
  <i />
</span>
`;

exports[`should match snapshots 5`] = `
.c0 i {
  background: #ffffff;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
}

.c1 i {
  -webkit-animation-direction: alternate;
  animation-direction: alternate;
  -webkit-animation-duration: 0.5s;
  animation-duration: 0.5s;
  -webkit-animation-fill-mode: none;
  animation-fill-mode: none;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-name: lfUrxy;
  animation-name: lfUrxy;
  -webkit-animation-play-state: running;
  animation-play-state: running;
  -webkit-animation-timing-function: ease-out;
  animation-timing-function: ease-out;
  border-radius: 100%;
  display: block;
  height: 6px;
  width: 6px;
  margin: 0 3px;
}

.c1 i:nth-child(1) {
  -webkit-animation-delay: 0.1s;
  animation-delay: 0.1s;
}

.c1 i:nth-child(2) {
  -webkit-animation-delay: 0.3s;
  animation-delay: 0.3s;
}

.c1 i:nth-child(3) {
  -webkit-animation-delay: 0.5s;
  animation-delay: 0.5s;
}

<span
  class="c0 c1 ButtonLoader primary"
>
  <i />
  <i />
  <i />
</span>
`;

exports[`should match snapshots 6`] = `
.c0 i {
  background: #a0a2ad;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
}

.c1 i {
  -webkit-animation-direction: alternate;
  animation-direction: alternate;
  -webkit-animation-duration: 0.5s;
  animation-duration: 0.5s;
  -webkit-animation-fill-mode: none;
  animation-fill-mode: none;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-name: lfUrxy;
  animation-name: lfUrxy;
  -webkit-animation-play-state: running;
  animation-play-state: running;
  -webkit-animation-timing-function: ease-out;
  animation-timing-function: ease-out;
  border-radius: 100%;
  display: block;
  height: 6px;
  width: 6px;
  margin: 0 3px;
}

.c1 i:nth-child(1) {
  -webkit-animation-delay: 0.1s;
  animation-delay: 0.1s;
}

.c1 i:nth-child(2) {
  -webkit-animation-delay: 0.3s;
  animation-delay: 0.3s;
}

.c1 i:nth-child(3) {
  -webkit-animation-delay: 0.5s;
  animation-delay: 0.5s;
}

<span
  class="c0 c1 ButtonLoader"
>
  <i />
  <i />
  <i />
</span>
`;

exports[`should match snapshots 7`] = `
.c0 i {
  background: #b7bac7;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
}

.c1 i {
  -webkit-animation-direction: alternate;
  animation-direction: alternate;
  -webkit-animation-duration: 0.5s;
  animation-duration: 0.5s;
  -webkit-animation-fill-mode: none;
  animation-fill-mode: none;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-name: lfUrxy;
  animation-name: lfUrxy;
  -webkit-animation-play-state: running;
  animation-play-state: running;
  -webkit-animation-timing-function: ease-out;
  animation-timing-function: ease-out;
  border-radius: 100%;
  display: block;
  height: 6px;
  width: 6px;
  margin: 0 3px;
}

.c1 i:nth-child(1) {
  -webkit-animation-delay: 0.1s;
  animation-delay: 0.1s;
}

.c1 i:nth-child(2) {
  -webkit-animation-delay: 0.3s;
  animation-delay: 0.3s;
}

.c1 i:nth-child(3) {
  -webkit-animation-delay: 0.5s;
  animation-delay: 0.5s;
}

<span
  class="c0 c1 ButtonLoader"
>
  <i />
  <i />
  <i />
</span>
`;
